docs: add client configuration section and support matrix for AI Bridge (#20640)
coder/coder: docs/ai-coder/ai-bridge.md

- `CODER_AIBRIDGE_OPENAI_BASE_URL` or `--aibridge-openai-base-url`

The default base URL (`https://api.openai.com/v1/`) works for the native OpenAI service. Point the base URL at your preferred OpenAI-compatible endpoint (for example, a hosted proxy or LiteLLM deployment) when needed.

#### Anthropic

Set the following when routing Anthropic-compatible traffic through Bridge:

- `CODER_AIBRIDGE_ANTHROPIC_KEY` or `--aibridge-anthropic-key`
- `CODER_AIBRIDGE_ANTHROPIC_BASE_URL` or `--aibridge-anthropic-base-url`

The default base URL (`https://api.anthropic.com/`) targets Anthropic's public API. Override it for Anthropic-compatible brokers.

##### Amazon Bedrock

Set the following when routing Amazon Bedrock traffic through Bridge:

- `CODER_AIBRIDGE_BEDROCK_REGION` or `--aibridge-bedrock-region`
- `CODER_AIBRIDGE_BEDROCK_ACCESS_KEY` or `--aibridge-bedrock-access-key`
- `CODER_AIBRIDGE_BEDROCK_ACCESS_KEY_SECRET` or `--aibridge-bedrock-access-key-secret`
- `CODER_AIBRIDGE_BEDROCK_MODEL` or `--aibridge-bedrock-model`
- `CODER_AIBRIDGE_BEDROCK_SMALL_FAST_MODEL` or `--aibridge-bedrock-small-fast-model`

#### Additional providers and Model Proxies

> Bridge can relay traffic to other OpenAI- or Anthropic-compatible services or model proxies like LiteLLM by pointing the base URL variables above at the provider you operate. Share feedback or follow along in the [`aibridge`](https://github.com/coder/aibridge) issue tracker as support for these providers stabilizes.

Reviewer: ```suggestion
Bridge can relay traffic to other OpenAI- or Anthropic-compatible services or model proxies like LiteLLM by pointing the base URL variables above at the provider you operate. Share feedback or follow along in the [`aibridge`](https://github.com/coder/aibridge) issue tracker as we expand support for additional providers.
```

Changed the last sentence to not call out "stabilizes". It implies that it's currently unstable and I'm not a fan of that implication

    ANTHROPIC_AUTH_TOKEN : data.coder_workspace_owner.me.session_token
  }
  ... # other agent configuration
}

# See https://registry.coder.com/modules/coder/claude-code for more information
module "claude-code" {
  count               = local.has_ai_prompt ? data.coder_workspace.me.start_count : 0
  source              = "dev.registry.coder.com/coder/claude-code/coder"
  version             = ">= 3.2.0"
  agent_id            = coder_agent.dev.id
  workdir             = "/home/coder/project"
  order               = 999
  claude_api_key      = data.coder_workspace_owner.me.session_token # To Enable AI Bridge integration
  ai_prompt           = data.coder_parameter.ai_prompt.value
  ... # other claude-code configuration
}

```

The same approach can be applied to pre-configure additional AI coding assistants by updating the base URL and API key settings.

#### Generic API key generation

Users can generate a Coder API key using either the CLI or the web UI. Follow the instructions at [Sessions and API tokens](../admin/users/sessions-tokens.md#generate-a-long-lived-api-token-on-behalf-of-yourself) to generate a Coder API key.

### Tested clients

> The combinations below reflect what we have exercised so far. Use the upstream links for vendor-specific steps to point each client at Bridge. Share additional findings in the [`aibridge`](https://github.com/coder/aibridge) issue tracker so we can keep this table current.

Reviewer: ```suggestion
The table below shows tested AI clients and their compatibility with AI Bridge. Click each client name for vendor-specific configuration instructions. Report issues or share compatibility updates in the [aibridge](https://github.com/coder/aibridge) issue tracker.
```
Changed up the phrasing to be more assertive/we've actually done the work.
